---
title: Long-term nutrient enrichment alters coastal plain wetland plant-microbe relationships
author: "Daniya Stephens, Aied Garcia, Jasper Leavitt, Regina Bledsoe, Carol Goodwillie, Ariane L. Peralta"
date: "Last updated on `r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    df_print: paged
  fig_caption: yes
  pdf_document: null
header-includes:
- \usepackage{array}
- \usepackage{graphics}
- \usepackage[utf8]{inputenc}
---

Project Description: Long-term nutrient enrichment and hydrology interact to modify plant-microbe interactions and community structures in a wetland ecosystem. 

------------------------------------------------------------------------------------------------------------------

This study was conduction at East Carolina Universityâ€™s West Research Campus (Greenville, North Carolina, USA), where a long-term nutrient addition (N-P-K fertilizers) and disturbance (by mowing) experiment was established in 2003.

This repository contains open-source code, data, & text files.

## We address the following questions:
* **Aim 1.)**: How does fertilization and hydrology influence the bacterial community composition change over time? 

* **Aim 2.)**: How does the fertilization and hydrology affect the relationship between plant and bacterial community composition change over time?

* **Aim 3.)**: How does the soil physico-chemical properties change over time?

### Repository Contents

* **analyses:** An R Markdown file (WRCforManuscript.Rmd) that includes R script written by Daniya Stephens and Jasper Leavitt containing functions used in analysis of soil and microbial sequence data.

* **data:** Files associated with soil and microbial data sets. 

* **figures:** Figures (of plant community composition) generated according to R script located in R Markdown file.

* **mothur:** Files containing script for bioinformatic analysis of 16S rRNA gene sequences (Illumina MiSeq platform) using mothur pipeline.

## Funding Sources

---------------------------------------------------------------------------------------------------------------------
*INSTALL PACKAGES*
```{r Initial Setup, include=FALSE}
rm(list=ls()) #clears history
setwd("~/GitHub/WRC2014to2020_PlantMicro/analyses") #sets working directory

#install.packages("ggplot2")
#install.packages("tidyverse")
#install.packages("dplyr")
#install.packages("reshape")
#install.packages("readr")
#install.packages("data.table")
#install.packages("phyloseq")
#install.packages("glmmTMB")
#install.packages("TMB")
#install.packages("BiocManager")
#BiocManager::install("phyloseq")
#source("https://raw.githubusercontent.com/joey711/phyloseq/master/inst/scripts/installer.R", local = TRUE)

#PLANT DIVERSITY FUNCTIONS and GRAPHING FUNCTIONS
require("ggplot2")
require("tidyverse")
require("dplyr")
require("reshape")

#BACTERIAL DIVERSITY FUNCTIONS
require("vegan")
require("ggplot2")
require("reshape2")
require("phyloseq")

#Code Dependencies
source("../bin/DiversityFunctions.R")
source("../bin/MothurTools.R")

#STATISTICAL FUNCTIONS
require("lme4")
require("agricolae")
require("car")
require("ade4")
require("glmmTMB")
```
---------------------------------------------------------------------------------------------------------------------
# Import RAW Plant Data and Calculate Importance Values
For analysis, percent cover data is combined with stem counts to calculate an importance value
```{r calc_importance,echo=FALSE, eval=FALSE}
WRC <- read.csv("../data/Plant_WRC_all.csv")
head(WRC)
str(WRC)
#WRC$SUM<-rowSums(WRC[,-1:-8])

WRC_norm <- WRC
for(i in 1:nrow(WRC)){
  WRC_norm[i,9:89] <- WRC[i,9:89]/sum(WRC[i,9:89])
}
head(WRC_norm)
str(WRC_norm)
which(apply(WRC_norm[,9:89],1,sum,na.rm=T)==0)

cover<-droplevels(subset(WRC_norm,data_type=="p cover"))
stem<-droplevels(subset(WRC_norm,data_type!="p cover"))
imp_values<-cover[,9:89]+stem[,9:89]
WRC_dat<-data.frame(cover[,1:8],imp_values)
head(WRC_dat)

write.csv(WRC_dat, file="../data/WRC_Importance.csv")
```

*READ IN THE PLANT FILE (subset 2014 to 2020) AND EXP DESIGN* 
```{r Import Files - Plants, include = FALSE}
file_path <- ("../data/WRC_Importance.csv") 
plant <- read.csv(file_path, header=TRUE) #Missing 2014, 2019, 2020 from data set

plant$fertilizer <- as.factor(plant$fertilizer)
plant$mowing <- as.factor(plant$mowing)
plant$ditch <- as.factor(plant$ditch)
str(plant)

#plant_15161718 <- subset(plant, year=="2015" | year=="2016" | year=="2017" | year=="2018" )
plant_1518 <- subset(plant, year=="2015" | year=="2018")
plant_M_MF <- subset(plant_1518, treatment == "M" | treatment == "MF")
#plant_M_MF <- subset(plant_15161718, treatment == "M" | treatment == "MF")

plant_M_MF_sort <- 
  plant_M_MF %>%
  group_by(year, treatment)
```

*READ SOIL AND DESIGN FILE*
```{r Import Files - Soil Properties, include = FALSE }
#file_path <- ("../data/WRC_2014to2020_design.csv")
file_path <- ("../data/WRC_2014_2015_2018_design_env.csv") 
soil_design <- read.csv(file_path)

#Convert numeric variable integer into a categorical variable 
soil_design$fertilization <- as.factor(soil_design$fertilization)
soil_design$mowing <- as.factor(soil_design$mowing)
soil_design$ditch <- as.factor(soil_design$ditch)
str(soil_design)

soil_design_M_MF <- subset(soil_design, treatment == "M" | treatment == "MF")

soil_design_M_MF_sort <- 
  soil_design_M_MF %>%
  group_by(year, treatment)
```

#BACTERIA
*READ IN BACTERIAL SEQUENCING FILES*
```{r Import Files - Bacteria, include= FALSE}
#READ IN OTU FILE

otu_file <-("../data/WRC2014to2020.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.shared")
otu <- read.otu(otu_file)

#READ IN TAXONOMY FILE

tax_file <- ("../data/WRC2014to2020.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.opti_mcc.0.03.cons.taxonomy")
tax <- import_mothur(mothur_constaxonomy_file =  tax_file)
colnames(tax)=c("Domain","Phylum","Class", "Order","Family","Genus")
tax_df <- as.data.frame(tax)
```
---------------------------------------------------------------------------------------------------------------------
*REMOVING SINGLETONS* 
```{r}
otu.2 <- otu[, which(colSums(otu) > 2)]
``` 
*TRANSFORM DATA*
```{r}
#RAREFY
min.N <- min(rowSums(otu.2))
otu_rare <- rrarefy(otu.2, min.N)

#RELATIVE ABUNDANCE
otu_rel <- otu_rare

for(i in 1:dim(otu_rare)[1]){
otu_rel[i,] <- otu_rare[i,]/sum(otu_rare[i,])
}
```
------------------------------------------------------------------------------------------------------
*ADD DIVERSITY METRICS TO bac_design*
```{r Create dataframe - Bacteria, include = FALSE}
#Renamed soil properties and design file
bac_design <- soil_design

#TRANSFORM BACTERIA DATA TO PRESENCE/ABCENSE
bc_PA <- decostand(otu_rare, method = "pa", na.rm=TRUE) #Missing WRC17_129

#Bacterial richness = NUMBER OF PRESENCE (1)
bc_richness <- specnumber(bc_PA) 
bc_richness

#Bacterial Shannon
bc_shannon <- diversity(otu_rel, "shannon")
bc_shannon

#Simpson's D Index
bc_invsimp <-diversity(otu_rel, "invsimpson")
bc_invsimp

#COMBINE bc_richness, bc_shannon, bc_invsimp
bc_diversity <- cbind(bac_design, bc_richness)
bc_diversity <- cbind(bc_diversity, bc_shannon)
bc_diversity <- cbind(bc_diversity, bc_invsimp)
str(bc_diversity)

#Added new column
bc_diversity$pH_avg = NA
#Calculate AVG pH 
bc_diversity$pH_avg = rowMeans(bc_diversity[,c(8:10)])
bc_diversity
```
*Diversity Metrics-- Richness, Shannon, Simpsons D*
```{r Diversity Metrics - Bacteria, echo = TRUE}
colors <- c("M" = "gray70", "MF" = "darkgreen","M" = "gray70", "MF" = "darkgreen","M" = "gray70", "MF" = "darkgreen") #SET COLORS FOR TREATMENTS

#Filter out Mowed and Mowed/Fertilized only
M_Mf <- bc_diversity[bc_diversity$treatment %in% c("M", "MF"), ]  

#RICHNESS
rich <- ggplot(M_Mf, aes(x=treatment, y=bc_richness, color=as.factor(treatment)))+ geom_boxplot()+
geom_point(aes(color=factor(treatment), shape=ditch), size=2, position = position_jitterdodge())+ scale_color_manual(name="treatment", values=c("gray70", "darkgreen"), labels = c("Mowed/Unfertilized", "Mowed/Fertilized"))+
xlab("Treatment")+
ylab("Bacterial OTU Richness")+
facet_wrap(~year)+
scale_shape_manual(labels = c("(0_wet)","(1_dry)"), values= c(16,17))+
theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm"))+ 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1))
rich
#Save plot 
ggsave("../figures/bc_richness.png", plot=rich, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)

#SHANNON
shannon <- ggplot(M_Mf, aes(x=treatment, y=bc_shannon, color=as.factor(treatment)))+ geom_boxplot() +
geom_point(aes(color=factor(treatment), shape=ditch), size=2, position = position_jitterdodge()) + scale_color_manual(name="treatment", values=c("gray70", "darkgreen"), labels = c("Mowed/Unfertilized", "Mowed/Fertilized"))+
xlab("Treatment")+
ylab("Shannon Diversity (H')")+
facet_wrap(~year)+
scale_shape_manual(labels = c("(0_wet)","(1_dry)"), values= c(16,17))+
theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm"))+ 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1))
shannon
#Save plot 
ggsave("../figures/bc_shannon.png", plot=shannon, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)

#SIMPSON D
simp_even <- ggplot(M_Mf, aes(x=treatment, y=bc_invsimp, color=as.factor(treatment)))+ geom_boxplot() +
geom_point(aes(color=factor(treatment), shape=ditch), size=2, position = position_jitterdodge()) + scale_color_manual(name="treatment", values=c("gray70", "darkgreen"), labels = c("Mowed/Unfertilized", "Mowed/Fertilized"))+
xlab("Treatment")+
ylab("Simpsons Index (D')")+
facet_wrap(~year)+
scale_shape_manual(labels = c("(0_wet)","(1_dry)"), values= c(16,17))+
theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm"))+ 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1))
simp_even
#Save plot 
ggsave("../figures/bc_evenness.png", plot=simp_even, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)
```
*Soil physicochemical properties*
```{r Plot - pH, carbon percent, nitrogen percent, echo= TRUE}
#Added new column
M_Mf$pH_avg = NA
#Calculate AVG pH 
M_Mf$pH_avg = rowMeans(M_Mf[,c(8:10)])
M_Mf

#pH-- Need to calculate the avg and use that number
p_pH <- ggplot(M_Mf, aes(x=treatment, y=pH_avg, color=as.factor(treatment)))+ geom_boxplot() +
geom_point(aes(color=factor(treatment), shape=ditch), size=2, position = position_jitterdodge()) + scale_color_manual(name="treatment", values=c("gray70", "darkgreen"), labels = c("Mowed/Unfertilized", "Mowed/Fertilized"))+
xlab("Treatment")+
ylab("Soil pH")+
facet_wrap(~year)+
scale_shape_manual(labels = c("(0_wet)","(1_dry)"), values= c(16,17))+
theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm"))+ 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1))
p_pH

#Save plot 
ggsave("../figures/p_pH.png", plot=p_pH, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)

#c percent 
c_percent <- ggplot(M_Mf, aes(x=treatment, y=c_percent, color=as.factor(treatment)))+ geom_boxplot() +
geom_point(aes(color=factor(treatment), shape=ditch), size=2, position = position_jitterdodge()) + scale_color_manual(name="treatment", values=c("gray70", "darkgreen"), labels = c("Mowed/Unfertilized", "Mowed/Fertilized"))+
xlab("Treatment")+
ylab("Total Soil Carbon (%)")+
facet_wrap(~year)+
scale_shape_manual(labels = c("(0_wet)","(1_dry)"), values= c(16,17))+
theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm"))+ 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1))
c_percent

#Save plot 
ggsave("../figures/c_percent.png", plot=c_percent, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)

#n percent 
n_percent <- ggplot(M_Mf, aes(x=treatment, y=n_percent, color=as.factor(treatment)))+ geom_boxplot() +
geom_point(aes(color=factor(treatment), shape=ditch), size=2, position = position_jitterdodge()) + scale_color_manual(name="treatment", values=c("gray70", "darkgreen"), labels = c("Mowed/Unfertilized", "Mowed/Fertilized"))+
xlab("Treatment")+
ylab("Total Soil Nitrogen (%)")+
facet_wrap(~year)+
scale_shape_manual(labels = c("(0_wet)","(1_dry)"), values= c(16,17))+
theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm"))+ 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1))

n_percent

#Save plot 
ggsave("../figures/n_percent.png", plot=n_percent, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)
```
*BRAY CURTIS DISSIMILARITY*
```{r Ordination (PCoA) - Bacteria, include= FALSE}
##This code builds the Bray Curtis dissimilarity matrix and accepts site-by-species matrix as input
bc_dist<-vegdist(otu_rel, method="bray")

bc_dist_pcoa <- cmdscale(bc_dist, k=3, eig=TRUE, add=FALSE)
##This code calculates the PCoA values that will be used to plot each sample on the PCoA graph.

###DO NOT EDIT
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
explainvar1a <- round(bc_dist_pcoa$eig[1] / sum(bc_dist_pcoa$eig), 3) * 100
explainvar2a <- round(bc_dist_pcoa$eig[2] / sum(bc_dist_pcoa$eig), 3) * 100
sum.eiga <- sum(explainvar1a, explainvar2a)
explainvar1a #(18.5%) #NEW DATA SHOWS 3.1%
explainvar2a #(12.5%) #NEW DATA SHOWS 2.7%
pcoa.groups <- paste(bac_design$ditch, bac_design$treatment, sep = "_") #These are factors from the design file.
pcoa.points <- data.frame(bc_dist_pcoa$points, group = pcoa.groups)  #df set in above chunk

###DO NOT EDIT BELOW HERE
#Set Std Err and Conf Int
se <- function(x, ...) {
sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))
}

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.groups <- paste(bc_diversity$ditch, bc_diversity$treatment, bc_diversity$year, sep = "_") #These are factors from the design file.
pcoa.points <- data.frame(bc_dist_pcoa$points, group = pcoa.groups)  #df set in above chunk


# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)


# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)
pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 2)) # treatment
pcoa.shape <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # source
pcoa.size <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 3))
levels(pcoa.col) <- c("Control","Fertilized","Mowed","Mowed-Fert")

df1a <- as.data.frame(pcoa.cent.dataframe)
plot1 <- ggplot(df1a, aes(x=V1, y=V2), group = interaction(pcoa.col, pcoa.shape, pcoa.size))+
theme_bw() +
geom_point(aes(shape = pcoa.shape, colour = pcoa.col, size = pcoa.size), stroke = 2) +
#geom_point(aes(colour=pcoa.col2), size =4.0)+
#removes gridlines from plot
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
axis.line = element_line(colour = "black"))+
#Set error barrs for geom_point
geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0), colour="black") + geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0), colour="black") +
#Set colors for treatments
scale_colour_manual(values = c("tan2", "plum1","gray70","darkgreen")) +
scale_shape_manual(values = c(16,22,15)) +
scale_size_manual(values = c(0.5,5,10)) +
#Sets map coordinates
coord_cartesian(xlim = c(-0.2, 0.2), ylim = c(-0.2, 0.2)) +
#Sets axis text and put border around plot
theme(axis.title = element_text(size=14),# face="bold"),
axis.text.x = element_text(size=14, color="black"),  axis.text.y = element_text(size=14, color="black"),
panel.border = element_rect(colour = "black", size=1.25)) +
#Set plot title textsize
theme(plot.title=element_text(size=14)) +
#Set legend text size
theme(legend.text=element_text(size=14, face="bold"), legend.title = element_text(size=14, face="bold"))+
#Sets size of tick marks on axis
theme(axis.ticks.length=unit(0.3,"cm")) +
#Sets labels for plot title, axis titles, and legend headings
xlab("PCoA 1 (18.5%)") + ylab("PCoA 2 (12.5%)") +
labs(shape = "Ditch") +
labs(colour="Treatment") +
ggtitle("Bacterial 16S rRNA Community Composition")+
theme(legend.position="right")
#View plot for all treatments
plot1



#ONLY MOWED AND MOWED-FERTILIZED
explainvar1a <- round(bc_dist_pcoa$eig[1] / sum(bc_dist_pcoa$eig), 3) * 100
explainvar2a <- round(bc_dist_pcoa$eig[2] / sum(bc_dist_pcoa$eig), 3) * 100
sum.eiga <- sum(explainvar1a, explainvar2a)
explainvar1a #(18.5%)
explainvar2a #(12.5%)
pcoa.groups <- paste(M_Mf$ditch, M_Mf$treatment, sep = "_") #These are factors from the design file.
pcoa.points <- data.frame(bc_dist_pcoa$points, group = pcoa.groups)  #df set in above chunk

###DO NOT EDIT BELOW HERE
#Set Std Err and Conf Int
se <- function(x, ...) {
sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))
}

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.groups <- paste(M_Mf$ditch, M_Mf$treatment, M_Mf$year, sep = "_") #These are factors from the design file.
pcoa.points <- data.frame(bc_dist_pcoa$points, group = pcoa.groups)  #df set in above chunk


# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)


# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)
pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 2)) # treatment
pcoa.shape <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # source
pcoa.size <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 3))
levels(pcoa.col) <- c("Mowed/Unfertilized","Mowed/Fertilized")


df1a <- as.data.frame(pcoa.cent.dataframe)
plot2 <- ggplot(df1a, aes(x=V1, y=V2), group = interaction(pcoa.col, pcoa.shape, pcoa.size))+
theme_bw() +
geom_point(aes(shape = pcoa.shape, colour = pcoa.col, size = pcoa.size), stroke = 2) +
#geom_point(aes(colour=pcoa.col2), size =4.0)+
#removes gridlines from plot
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
axis.line = element_line(colour = "black"))+
#Set error barrs for geom_point
geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0), colour="black") + 
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0), colour="black") +
#Set colors for treatments
scale_colour_manual(values = c("gray70","darkgreen","gray70","darkgreen")) +
scale_shape_manual(values = c(16,22,15)) +
  scale_size_manual(values = c(0.5,5,10)) +
#Sets map coordinates
coord_cartesian(xlim = c(-0.2, 0.2), ylim = c(-0.2, 0.2)) +
#Sets axis text and put border around plot
theme(axis.title = element_text(size=14),# face="bold"),
axis.text.x = element_text(size=14, color="black"),  axis.text.y = element_text(size=14, color="black"),
panel.border = element_rect(colour = "black", size=1.25)) +
#Set plot title textsize
theme(plot.title=element_text(size=14)) +
#Set legend text size
theme(legend.text=element_text(size=14, face="bold"), legend.title = element_text(size=14, face="bold"))+
#Sets size of tick marks on axis
theme(axis.ticks.length=unit(0.3,"cm")) +
#Sets labels for plot title, axis titles, and legend headings
xlab("PCoA 1 (18.5%)") + ylab("PCoA 2 (12.5%)") +
labs(shape = "Ditch") +
labs(colour="Treatment") +
ggtitle("Bacterial 16S rRNA Community Composition")+
theme(legend.position="right")

plot2

#Save plot 
ggsave("../figures/bc_PCoA_plot2.png", plot=plot2, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)
```
#UPDATE ACCORDING TO MODELS USED IN GOODWILLIE ET AL. 2020 PAPER
*Statistics* 
```{r anova and perMANOVA - Bacteria - All treatments, echo= TRUE}

#FROM GOODWILLIE PAPER
b1=glmmTMB(totalbiomass~Year*Treatment*Ditch+
              (1|Ditch/Block/Quadrat), data=biomass, family=gaussian(log))
library(DHARMa)
res=simulateResiduals(b1)
plot(res)

Anova(b1,type="II")

alphae <- allEffects(b1)
alphae
plot(alphae)
summary(b1)

#ALL TREATMENTS(Mowed/Unfertilized, Mowed/Fertilized, Unmowed/Unfertilized, Unmowed, Fertilized)

#Nitrogen content 
n_percent_aov <- aov(n_percent ~ fertilization*mowing*ditch*year, data = bc_diversity)
anova(n_percent_aov)

#Carbon content 
c_percent_aov <- aov(log(c_percent) ~ fertilization*mowing*ditch*year, data = bc_diversity)
 anova(c_percent_aov)

#pH
pH_aov <- aov(pH_avg ~ fertilization*mowing*ditch*year, data = bc_diversity)
anova(pH_aov)

#Bacterial community composition

#RICHNESS 
bcrichness_aov <- aov(bc_richness ~ fertilization*mowing*ditch*year, data = bc_diversity)
anova(bcrichness_aov)

#SHANNON
bcshannon_aov <- aov(bc_shannon ~ fertilization*mowing*ditch*year, data = bc_diversity)
anova(bcshannon_aov)

#INVSIMP
bcinvsimp_aov <- aov(bc_invsimp ~ fertilization*mowing*ditch*year, data = bc_diversity)
anova(bcinvsimp_aov)


#MOWED AND MOWED/FERT ONLY--Removed Mowing variable
bc_diversity_M_Mf <- bc_diversity[bc_diversity$treatment %in% c("M", "MF"),] 

#Nitrogen content 
n_percent_aov <- aov(n_percent ~ fertilization*ditch*year, data = bc_diversity_M_Mf)
anova(n_percent_aov)

#Carbon content 
c_percent_aov <- aov(log(c_percent) ~ fertilization*ditch*year, data = bc_diversity_M_Mf)
anova(c_percent_aov)

#pH
pH_aov <- aov(pH_avg ~ fertilization*ditch*year, data = bc_diversity_M_Mf)
anova(pH_aov)

#Bacterial community composition

#RICHNESS 
bcrichnessMMF_aov <- aov(bc_richness ~ fertilization*ditch*year, data = bc_diversity_M_Mf)
anova(bcrichnessMMF_aov)

#SHANNON
bcshannonMMF_aov <- aov(bc_shannon ~ fertilization*ditch*year, data = bc_diversity_M_Mf)
anova(bcshannonMMF_aov)

#INVSIMP
bcinvsimpMMF_aov <- aov(bc_invsimp ~ fertilization*ditch*year, data = bc_diversity_M_Mf)
anova(bcinvsimpMMF_aov)

#PERMANOVA -- ALL TREATMENTS
design_bac2 <- cbind(bc_diversity, otu_rel)

# Convert integers to characters for fertilization, mowing, and ditch only
design_bac2$fertilization <- as.factor(design_bac2$fertilization)
design_bac2$mowing <- as.factor(design_bac2$mowing)
design_bac2$ditch <- as.factor(design_bac2$ditch)

adonis.all <- adonis(design_bac2[,-c(1:18)] ~ fertilization*ditch*year*mowing, method = "bray", data = design_bac2, perm=1000, set.seed=42)

adonis.all

#PERMANOVA -- MOWED/UNFERTILIZED AND MOWED/FERTILIZED--Removed mowing variable

design_bac3 <- subset(design_bac2, treatment == "M" | treatment == "MF")

# Convert integers to characters
design_bac3$fertilization <- as.factor(design_bac3$fertilization)
design_bac3$mowing <- as.factor(design_bac3$mowing)
design_bac3$year <- as.factor(design_bac3$year)

adonis.M_MF <- adonis(design_bac3[,-c(1:18)] ~ fertilization*ditch*year, method = "bray", data = design_bac3, perm=1000, set.seed=42)
adonis.M_MF
```
-------------------------------------------------------------------------------------------------------
#PLANTS (ALL YEARS)

*Diversity Metrics-- Richness, Shannon, Simpson's D*
```{r Diversity Metrics - Plant, echo= TRUE}
#CALCULATE PLANT RICHNESS
str(plant_M_MF_sort)
plant_sub <- data.frame(plant_M_MF_sort[,-c(1:9)])
plantdesign <- data.frame(plant_M_MF_sort[,c(1:9)])

#TRANSFORM DATA TO 0/1 (ABSENCE/PRESENCE)
plant_PA <- decostand(plant_sub, method = "pa", na.rm=TRUE)

#COUNT THE NUMBER OF PRESENT(1) IN THE SAMPLE
plant_richness <- specnumber(plant_PA)
plant_richness

#CALCULATE SHANNON H'
plant_shannon <- diversity(plant_sub, "shannon")
plant_shannon
 
#CALCULATING INVERSE SIMPSONS D
plant_invsimp <-diversity(plant_sub, "invsimpson")
plant_invsimp

#COMBINE plant_richness, plant_shannon, plant_invsimp
plant_diversity <- cbind(plantdesign, plant_richness)
plant_diversity <- cbind(plant_diversity, plant_shannon)
plant_diversity <- cbind(plant_diversity, plant_invsimp)

str(plant_diversity)

#PLOT PLANT RICHNESS
plant_rich <- ggplot(plant_diversity, aes(x=treatment, y=plant_richness, color=as.factor(treatment)))+ geom_boxplot()+
geom_point(aes(color=factor(treatment), shape=ditch), size=2, position = position_jitterdodge())+ scale_color_manual(name="treatment", values=c("gray70", "darkgreen"), labels = c("Mowed/Unfertilized", "Mowed/Fertilized"))+
facet_wrap(~year)+
xlab("Treatment")+
ylab("Plant Species Richness")+
scale_shape_manual(labels = c("(0_wet)","(1_dry)"), values= c(16,17))+
theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm"))+ 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1))
plant_rich

#Save plot 
ggsave("../figures/plant_richness.png", plot=plant_rich, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)


#PLOT SHANNON DIVERSITY
plant_shannon <- ggplot(plant_diversity, aes(x=treatment, y=plant_shannon, color=as.factor(treatment)))+ geom_boxplot()+
geom_point(aes(color=factor(treatment), shape=ditch), size=2, position = position_jitterdodge())+ scale_color_manual(name="treatment", values=c("gray70", "darkgreen"), labels = c("Mowed/Unfertilized", "Mowed/Fertilized"))+
facet_wrap(~year)+
xlab("Treatment")+
ylab("Shannon Diversity (H')")+
scale_shape_manual(labels = c("(0_wet)","(1_dry)"), values= c(16,17))+
theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm"))+ 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1))
plant_shannon

#Save plot 
ggsave("../figures/plant_shannon.png", plot=plant_shannon, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)


#PLOT SIMPSON EVENNESS
plant_invsimp <- ggplot(plant_diversity, aes(x=treatment, y=plant_invsimp, color=as.factor(treatment)))+ geom_boxplot()+
geom_point(aes(color=factor(treatment), shape=ditch), size=2, position = position_jitterdodge())+ scale_color_manual(name="treatment", values=c("gray70", "darkgreen"), labels = c("Mowed/Unfertilized", "Mowed/Fertilized"))+
facet_wrap(~year)+
xlab("Treatment")+
ylab("Simpson's Index (D')")+
scale_shape_manual(labels = c("(0_wet)","(1_dry)"), values= c(16,17))+
theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm"))+ 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1))
plant_invsimp

#Save plot 
ggsave("../figures/plant_evenness.png", plot=plant_invsimp, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)
```
*CALCULATE DISTANCE MATRIX AND PCoA COMPONENTS*
```{r Ordination (PCoA)- Plant, include= FALSE}
#BUILD THE MATRIX
plant_dist <-vegdist(plant_sub, method="bray")

#CALCULATE PCoA VALUES
plant_dist_pcoa <- cmdscale(plant_dist, k=3, eig=TRUE, add=FALSE)

#CLASSICAL (METRIC) MULTIDIMENSIONAL SCALING
explainvar1a <- round(plant_dist_pcoa$eig[1] / sum(plant_dist_pcoa$eig), 3) * 100
explainvar2a <- round(plant_dist_pcoa$eig[2] / sum(plant_dist_pcoa$eig), 3) * 100
sum.eiga <- sum(explainvar1a, explainvar2a)
explainvar1a #27.4%
explainvar2a #14.3%
pcoa.groups <- paste(plant_M_MF_sort$ditch, plant_M_MF_sort$treatment, plant_M_MF_sort$year, sep = "_") #These are factors from the design file.
pcoa.points <- data.frame(plant_dist_pcoa$points, group = pcoa.groups)  #df set in above chunk

#Set Std Err and Conf Int
se <- function(x, ...) {
  sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))
}


# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)


# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)
pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 2)) # treatment
pcoa.shape <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # source
pcoa.size <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 3))
levels(pcoa.col) <- c("Mowed/Unfertilized","Mowed/Fertilized")

df1a <- as.data.frame(pcoa.cent.dataframe)
plot2 <- ggplot(df1a, aes(x=V1, y=V2), group = interaction(pcoa.col, pcoa.shape, pcoa.size)) +
theme_bw() +
geom_point(aes(shape = pcoa.shape, colour = pcoa.col, size = pcoa.size), stroke = 2) +
#geom_point(aes(colour=pcoa.col2), size =4.0)+
#removes gridlines from plot
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
axis.line = element_line(colour = "black"))+
#Set error barrs for geom_point+
geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0), colour="black") + 
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0), colour="black") +
#Set colors for treatments
scale_colour_manual(values = c("gray70","darkgreen")) +
scale_shape_manual(values = c(16,22,15)) +
  scale_size_manual(values = c(0.5,5,10)) +
#Sets map coordinates
coord_cartesian(xlim = c(-0.2, 0.2), ylim = c(-0.2, 0.2)) +
#Sets axis text and put border around plot
theme(axis.title = element_text(size=14),# face="bold"),
axis.text.x = element_text(size=14, color="black"),  axis.text.y = element_text(size=14, color="black"),
panel.border = element_rect(colour = "black", size=1.25)) +
#Set plot title textsize
theme(plot.title=element_text(size=14)) +
#Set legend text size
theme(legend.text=element_text(size=14, face="bold"), legend.title = element_text(size=14, face="bold"))+
#Sets size of tick marks on axis
theme(axis.ticks.length=unit(0.3,"cm")) +
#Sets labels for plot title, axis titles, and legend headings
xlab("PCoA 1 (27.4%)") + ylab("PCoA 2 (14.3%)") +
labs(shape = "Ditch") +
labs(colour="Treatment") +
ggtitle("Plant Community Composition")+
theme(legend.position="right")

plot2

#Save plot 
ggsave("../figures/plant_PCoA_plot.png", plot=plot2, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)

plant.adonis.M_MF <- adonis(plant_M_MF_sort[,-c(1:9)] ~ fertilizer*ditch*year, method = "bray", data = plant_M_MF_sort, perm=1000, set.seed=42)
plant.adonis.M_MF

```
*Statistics* 
```{r anova and perMANOVA - Plants - All treatments, echo= TRUE}
#Plant community composition

#RICHNESS 
plantrichnessMMF_aov <- aov(plant_richness ~ fertilizer*ditch*year, data = plant_diversity)
anova(plantrichnessMMF_aov)

#SHANNON
plantshannonMMF_aov <- aov(plant_shannon ~ fertilizer*ditch*year, data = plant_diversity)
anova(plantshannonMMF_aov)

#INVSIMP
plantinvsimpMMF_aov <- aov(plant_invsimp ~ fertilizer*ditch*year, data = plant_diversity)
anova(plantinvsimpMMF_aov)
```





