---
title: "Updated WRC: Long-term Nutrient Enrichment and Hydrology Interact to Modify Plant-Microbe Interactions in a Wetland Ecosystem"
author: "Daniya Stephens, Jasper Leavitt"
date: "05/25/2021"
output: html_notebook
---
------------------------------------------------------------------------------------------------------------------

This study was conduction at East Carolina Universityâ€™s West Research Campus (Greenville, North Carolina, USA), where a long-term nutrient addition (N-P-K fertilizers) and disturbance (by mowing) experiment was established in 2003.

This repository contains open-source code, data, & text files.

## We address the following questions:
* **Aim 1.)**: How does fertilization and hydrology influence the bacterial community composition change over time? 

* **Aim 2.)**: How does the fertilization and hydrology affect the relationship between plant and bacterial community composition change over time?

### Repository Contents

* **analyses:** An R Markdown file (WRCforManuscript.Rmd) that includes R script written by Daniya Stephens and Jasper Leavitt containing functions used in analysis of soil and microbial sequence data.

* **data:** Files associated with soil and microbial data sets. 

* **figures:** Figures (of plant community composition) generated according to R script located in R Markdown file.

* **mothur:** Files containing script for bioinformatic analysis of 16S rRNA gene sequences (Illumina MiSeq platform) using mothur pipeline.

## Funding Sources

---------------------------------------------------------------------------------------------------------------------
*INSTALL PACKAGES*
```{r Initial Setup, include=FALSE}
#PLANT DIVERSITY FUNCTIONS
install.packages("ggplot2")
install.packages("tidyverse")
install.packages("dplyr")
install.packages("reshape")
install.packages("readr")
library("ggplot2")
library("tidyverse")
require("dplyr")
require("reshape")

#BACTERIAL DIVERSITY FUNCTIONS
require("vegan")
require("ggplot2")
require("reshape2")
require("phyloseq")
MothurTools <- ("C:/Users/ohhey/OneDrive/Desktop/WRC2020/MothurTools.R")
source("C:/Users/ohhey/OneDrive/Desktop/WRC2020/MothurTools.R")

#STATISTICAL FUNCTIONS
require("lme4")
require("agricolae")
require("car")
require("ade4")

install.packages("data.table")
install.packages("phyloseq")
install.packages("BiocManager")
BiocManager::install("phyloseq")
source("https://raw.githubusercontent.com/joey711/phyloseq/master/inst/scripts/installer.R", local = TRUE)
```
---------------------------------------------------------------------------------------------------------------------
*READ IN THE PLANT FILE (2015 & 2018) AND EXP DESIGN* 
```{r Import Files - Plants, include = FALSE}

file_path <- ("C:/Users/ohhey/OneDrive/Desktop/WRCcsvfiles/WRC1518_plantdata.csv") 
plant <- read.csv(file_path)
plant_M_MF <- subset(plant, treatment == "M" | treatment == "MF")

#plant_design <- ("C:/Users/ohhey/OneDrive/Desktop/WRCcsvfiles/2015_2018_WRC_soil_properties.csv")
#df_plantdesign <- read.csv(plant_design)
```
*READ IN 2015_2018_WRC_soil_properties*
```{r Import Files - Soil Properties, include = FALSE }
file_path <- ("C:/Users/ohhey/OneDrive/Desktop/WRCcsvfiles/2015_2018_WRC_soil_properties.csv")
soil <- read.csv(file_path)

#Subset source and treatment 
soil_M_MF <- subset(soil, source == "bulk")
soil_M_MF <- subset(soil_M_MF, treatment == "M" | treatment == "MF")

plant_M_MF_sort <- 
  plant_M_MF %>%
  group_by(year, treatment)

soil_M_MF_sort <- 
  soil_M_MF %>%
  group_by(year, treatment)
```
*READ IN BACTERIAL DIVERSITY FILES*
```{r Import Files - Bacteria, include= FALSE}
#READ IN OTU FILE

otu_file <-("C:/Users/ohhey/OneDrive/Desktop/WRCcsvfiles/WRC141518.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.shared")
otu <- read.otu(otu_file)

#READ IN TAXONOMY FILE

tax_file <- ("C:/Users/ohhey/OneDrive/Desktop/WRCcsvfiles/WRC141518.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.pick.opti_mcc.0.03.cons.taxonomy")
tax <- import_mothur(mothur_constaxonomy_file =  tax_file)
colnames(tax)=c("Domain","Phylum","Class", "Order","Family","Genus")
tax_df <- as.data.frame(tax)

#READ IN CSV FILE AS bac_design

file_path <- ("C:/Users/ohhey/OneDrive/Desktop/WRCcsvfiles/WRC_2014_2015_2018_design_env.csv")
bac_design <- read.csv(file_path)
```
---------------------------------------------------------------------------------------------------------------------
*REMOVING SEQUENCING ERRORS* 
```{r}
otu.2 <- otu[, which(colSums(otu) > 2)]
#otu.10 <- otu[, which(colSums(otu) > 9)]
``` 
*TRANSFORM DATA*
```{r}
#RAREFY

min.N <- min(rowSums(otu.2))
otu_rare <- rrarefy(otu.2, min.N)

#RELATIVE ABUNDANCE
otu_rel <- otu_rare

for(i in 1:dim(otu_rare)[1]){
otu_rel[i,] <- otu_rare[i,]/sum(otu_rare[i,])
}

#Convert numeric variable integer into a categorical variable 
bac_design$block <- as.factor(bac_design$block) 
bac_design$ditch <- as.factor(bac_design$ditch)
```
-------------------------------------------------------------------------------------------------------
#BACTERIA

*TRANSFORM BACTERIA DATA TO PRESENCE/ABCENSE*
```{r}
bc_PA <- decostand(otu_rare, method = "pa", na.rm=TRUE) 

#NUMBER OF PRESENCE (1)
bc_richness <- specnumber(bc_PA) 
bc_richness

#SAMPLES WITH THE HIGHEST AND LOWEST RICHNESS
rich<- colSums(t(bc_richness))
rich[which.min(colSums(t(bc_richness)))]  #MIN- WRC15_38 (1918)
rich[which.max(colSums(t(bc_richness)))]  #MAX- WRC18_192(2697)
```
*ADD DIVERSITY METRICS TO bac_design*
```{r Create dataframe - Bacteria, include = FALSE}
#Bacterial richness 
bc_diversity <- cbind(bac_design, bc_richness)

str(bc_diversity)
bc_diversity$ditch <- as.factor(bc_diversity$ditch)

#Bacterial Shannon
bc_shannon <- diversity(otu_rel, "shannon")
bc_shannon

#COMBINE bc_diversity and bc_shannon
bc_diversity <- cbind(bc_diversity, bc_shannon)
bc_diversity$ditch <- as.factor(bc_diversity$ditch)

#Simpson's D Index
bc_invsimp <-diversity(otu_rel, "invsimpson")
bc_invsimp

#ADD bc_invsimp TO bc_diversity
bc_diversity <- cbind(bc_diversity, bc_invsimp)
str(bc_diversity)
bc_diversity$ditch <- as.factor(bc_diversity$ditch)

#Filter out Mowed and Mowed/Fertilized only
M_Mf <- bc_diversity[bc_diversity$treatment %in% c("M", "MF"), ]  
```
*Diversity Metrics-- Richness, Shannon, Simpsons D*
```{r Diversity Metrics - Bacteria, echo = TRUE}
colors <- c("M" = "gray70", "MF" = "darkgreen","M" = "gray70", "MF" = "darkgreen","M" = "gray70", "MF" = "darkgreen") #SET COLORS FOR TREATMENTS


#RICHNESS
rich <- ggplot(M_Mf, aes(x=treatment, y=bc_richness, color=as.factor(treatment)))+ geom_boxplot()+
geom_point(aes(color=factor(treatment), shape=ditch), size=2, position = position_jitterdodge())+ scale_color_manual(name="Treatment", values=c("gray70", "darkgreen"), labels = c("Mowed", "Mowed/Fertilized"))+
xlab("Treatment")+
ylab("Bacterial Richness (OTUs)")+
facet_wrap(~year)+
scale_shape_manual(labels = c("(0_wet)","(1_dry)"), values= c(16,17))+
theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm"))+ 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1))
#Save plot 
ggsave("C:/Users/ohhey/OneDrive/Pictures/WRC20141518/rich.png", plot=rich, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)

#SHANNON
shannon <- ggplot(M_Mf, aes(x=treatment, y=bc_shannon, color=as.factor(treatment)))+ geom_boxplot() +
geom_point(aes(color=factor(treatment), shape=ditch), size=2, position = position_jitterdodge()) + scale_color_manual(name="Treatment", values=c("gray70", "darkgreen"), labels = c("Mowed", "Mowed/Fertilized"))+
xlab("Treatment")+
ylab("Shannon Diversity (H')")+
facet_wrap(~year)+
scale_shape_manual(labels = c("(0_wet)","(1_dry)"), values= c(16,17))+
theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm"))+ 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1))
#Save plot 
ggsave("C:/Users/ohhey/OneDrive/Pictures/WRC20141518/shannon.png", plot=shannon, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)

#SIMPSON D
simp <- ggplot(M_Mf, aes(x=treatment, y=bc_invsimp, color=as.factor(treatment)))+ geom_boxplot() +
geom_point(aes(color=factor(treatment), shape=ditch), size=2, position = position_jitterdodge()) + scale_color_manual(name="Treatment", values=c("gray70", "darkgreen"), labels = c("Mowed", "Mowed/Fertilized"))+
xlab("Treatment")+
ylab("Simpsons Index (D)")+
facet_wrap(~year)+
scale_shape_manual(labels = c("(0_wet)","(1_dry)"), values= c(16,17))+
theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm"))+ 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1))
#Save plot 
ggsave("C:/Users/ohhey/OneDrive/Pictures/WRC20141518/simp.png", plot=simp, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)
```
*Soil physicochemical properties*
```{r Plot - pH, carbon percent, nitrogen percent, echo= TRUE}
#pH
p_pH <- ggplot(M_Mf, aes(x=treatment, y=pH, color=as.factor(treatment)))+ geom_boxplot() +
geom_point(aes(color=factor(treatment), shape=ditch), size=2, position = position_jitterdodge()) + scale_color_manual(name="Treatment", values=c("gray70", "darkgreen"), labels = c("Mowed", "Mowed/Fertilized"))+
xlab("Treatment")+
ylab("pH")+
facet_wrap(~year)+
scale_shape_manual(labels = c("(0_wet)","(1_dry)"), values= c(16,17))+
theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm"))+ 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1))
p_pH

#Save plot 
ggsave("C:/Users/ohhey/OneDrive/Pictures/p_pH.png", plot=p_pH, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)

#c percent 
c_percent <- ggplot(M_Mf, aes(x=treatment, y=c_percent, color=as.factor(treatment)))+ geom_boxplot() +
geom_point(aes(color=factor(treatment), shape=ditch), size=2, position = position_jitterdodge()) + scale_color_manual(name="Treatment", values=c("gray70", "darkgreen"), labels = c("Mowed", "Mowed/Fertilized"))+
xlab("Treatment")+
ylab("Total Carbon Percent (%)")+
facet_wrap(~year)+
scale_shape_manual(labels = c("(0_wet)","(1_dry)"), values= c(16,17))+
theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm"))+ 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1))
c_percent

#Save plot 
ggsave("C:/Users/ohhey/OneDrive/Pictures/c_percent.png", plot=c_percent, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)

#another visual
ggplot(bac_design, aes(treatment, c_percent)) +
  geom_point()+
  facet_wrap(~ditch + year)


#n percent 
n_percent <- ggplot(M_Mf, aes(x=treatment, y=n_percent, color=as.factor(treatment)))+ geom_boxplot() +
geom_point(aes(color=factor(treatment), shape=ditch), size=2, position = position_jitterdodge()) + scale_color_manual(name="Treatment", values=c("gray70", "darkgreen"), labels = c("Mowed", "Mowed/Fertilized"))+
xlab("Treatment")+
ylab("Total Nitrogen Percent (%)")+
facet_wrap(~year)+
scale_shape_manual(labels = c("(0_wet)","(1_dry)"), values= c(16,17))+
theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm"))+ 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1))


n_percent

#Save plot 
ggsave("C:/Users/ohhey/OneDrive/Pictures/WRC20141518/n_percent.png", plot=n_percent, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)
```
*BRAY CURTIS DISSIMILARITY*
```{r Ordination (PCoA) - Bacteria, include= FALSE}
##This code builds the Bray Curtis dissimilarity matrix and accepts site-by-species matrix as input
bc_dist<-vegdist(otu_rel, method="bray")

bc_dist_pcoa <- cmdscale(bc_dist, k=3, eig=TRUE, add=FALSE)
##This code calculates the PCoA values that will be used to plot each sample on the PCoA graph.

###DO NOT EDIT
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
explainvar1a <- round(bc_dist_pcoa$eig[1] / sum(bc_dist_pcoa$eig), 3) * 100
explainvar2a <- round(bc_dist_pcoa$eig[2] / sum(bc_dist_pcoa$eig), 3) * 100
sum.eiga <- sum(explainvar1a, explainvar2a)
explainvar1a #(18.6%)
explainvar2a #(12.5%)
pcoa.groups <- paste(bac_design$ditch, bac_design$treatment, sep = "_") #These are factors from the design file.
pcoa.points <- data.frame(bc_dist_pcoa$points, group = pcoa.groups)  #df set in above chunk

###DO NOT EDIT BELOW HERE
#Set Std Err and Conf Int
se <- function(x, ...) {
sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))
}

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.groups <- paste(bc_diversity$ditch, bc_diversity$treatment, bc_diversity$year, sep = "_") #These are factors from the design file.
pcoa.points <- data.frame(bc_dist_pcoa$points, group = pcoa.groups)  #df set in above chunk


# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)


# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)
pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 2)) # treatment
pcoa.shape <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # source
pcoa.size <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 3))
levels(pcoa.col) <- c("Control","Fertilized","Mowed","Mowed-Fert")


df1a <- as.data.frame(pcoa.cent.dataframe)
plot2 <- ggplot(df1a, aes(x=V1, y=V2), group = interaction(pcoa.col, pcoa.shape, pcoa.size)) +
theme_bw() +
geom_point(aes(shape = pcoa.shape, colour = pcoa.col, size = pcoa.size), stroke = 2) +
#geom_point(aes(colour=pcoa.col2), size =4.0)+
#removes gridlines from plot
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
axis.line = element_line(colour = "black"))+
#Set error barrs for geom_point+
geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0), colour="black") + geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0), colour="black") +
#Set colors for treatments
scale_colour_manual(values = c("tan2", "plum1","gray70","darkgreen")) +
scale_shape_manual(values = c(16,22,15)) +
  scale_size_manual(values = c(0.5,5,10)) +
#Sets map coordinates
coord_cartesian(xlim = c(-0.2, 0.2), ylim = c(-0.2, 0.2)) +
#Sets axis text and put border around plot
theme(axis.title = element_text(size=14),# face="bold"),
axis.text.x = element_text(size=14, color="black"),  axis.text.y = element_text(size=14, color="black"),
panel.border = element_rect(colour = "black", size=1.25)) +
#Set plot title textsize
theme(plot.title=element_text(size=14)) +
#Set legend text size
theme(legend.text=element_text(size=14, face="bold"), legend.title = element_text(size=14, face="bold"))+
#Sets size of tick marks on axis
theme(axis.ticks.length=unit(0.3,"cm")) +
#Sets labels for plot title, axis titles, and legend headings
xlab("PCoA 1 (18.6%)") + ylab("PCoA 2 (12.5%)") +
labs(shape = "Ditch") +
labs(colour="Treatment") +
ggtitle("16S rRNA Community Diversity")+
theme(legend.position="right")

plot2



#ONLY MOWED AND MOWED-FERTILIZED
explainvar1a <- round(bc_dist_pcoa$eig[1] / sum(bc_dist_pcoa$eig), 3) * 100
explainvar2a <- round(bc_dist_pcoa$eig[2] / sum(bc_dist_pcoa$eig), 3) * 100
sum.eiga <- sum(explainvar1a, explainvar2a)
explainvar1a #(18.6%)
explainvar2a #(12.5%)
pcoa.groups <- paste(M_Mf$ditch, M_Mf$treatment, sep = "_") #These are factors from the design file.
pcoa.points <- data.frame(bc_dist_pcoa$points, group = pcoa.groups)  #df set in above chunk

###DO NOT EDIT BELOW HERE
#Set Std Err and Conf Int
se <- function(x, ...) {
sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))
}

# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.groups <- paste(M_Mf$ditch, M_Mf$treatment, M_Mf$year, sep = "_") #These are factors from the design file.
pcoa.points <- data.frame(bc_dist_pcoa$points, group = pcoa.groups)  #df set in above chunk


# Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)


# Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)
pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 2)) # treatment
pcoa.shape <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # source
pcoa.size <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 3))
levels(pcoa.col) <- c("Mowed","Mowed-Fert")


df1a <- as.data.frame(pcoa.cent.dataframe)
plot2 <- ggplot(df1a, aes(x=V1, y=V2), group = interaction(pcoa.col, pcoa.shape, pcoa.size)) +
theme_bw() +
geom_point(aes(shape = pcoa.shape, colour = pcoa.col, size = pcoa.size), stroke = 2) +
#geom_point(aes(colour=pcoa.col2), size =4.0)+
#removes gridlines from plot
theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
axis.line = element_line(colour = "black"))+
#Set error barrs for geom_point+
geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0), colour="black") + geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0), colour="black") +
#Set colors for treatments
scale_colour_manual(values = c("gray70","darkgreen")) +
scale_shape_manual(values = c(16,22,15)) +
  scale_size_manual(values = c(0.5,5,10)) +
#Sets map coordinates
coord_cartesian(xlim = c(-0.2, 0.2), ylim = c(-0.2, 0.2)) +
#Sets axis text and put border around plot
theme(axis.title = element_text(size=14),# face="bold"),
axis.text.x = element_text(size=14, color="black"),  axis.text.y = element_text(size=14, color="black"),
panel.border = element_rect(colour = "black", size=1.25)) +
#Set plot title textsize
theme(plot.title=element_text(size=14)) +
#Set legend text size
theme(legend.text=element_text(size=14, face="bold"), legend.title = element_text(size=14, face="bold"))+
#Sets size of tick marks on axis
theme(axis.ticks.length=unit(0.3,"cm")) +
#Sets labels for plot title, axis titles, and legend headings
xlab("PCoA 1 (18.6%)") + ylab("PCoA 2 (12.5%)") +
labs(shape = "Ditch") +
labs(colour="Treatment") +
ggtitle("16S rRNA Community Diversity")+
theme(legend.position="right")

plot2

#Save plot 
ggsave("C:/Users/ohhey/OneDrive/Pictures/WRC20141518/plot2.png", plot=plot2, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)
```
*Statistics*
```{r anova and perMANOVA - Bacteria - All treatments, echo= TRUE}
#ALL TREATMENTS

#Nitrogen content 
n_percent_aov <- aov(n_percent ~ fertilization*mowing*ditch*year, data = bc_diversity)
anova(n_percent_aov)

#Carbon content 
c_percent_aov <- aov(log(c_percent) ~ fertilization*mowing*ditch*year, data = bc_diversity)
 anova(c_percent_aov)

#pH
pH_aov <- aov(pH ~ fertilization*mowing*ditch*year, data = bc_diversity)
anova(pH_aov)

#Bacterial community composition

#RICHNESS 
bcrichness_aov <- aov(bc_richness ~ fertilization*mowing*ditch*year, data = bc_diversity)
anova(bcrichness_aov)

#SHANNON
bcshannon_aov <- aov(bc_shannon ~ fertilization*mowing*ditch*year, data = bc_diversity)
anova(bcshannon_aov)

#INVSIMP
bcinvsimp_aov <- aov(bc_invsimp ~ fertilization*mowing*ditch*year, data = bc_diversity)
anova(bcinvsimp_aov)

#MOWED AND MOWED/FERT ONLY---- (data=M_F, have to find what the new data is called)

#Nitrogen content 
n_percent_aov <- aov(n_percent ~ fertilization*mowing*ditch*year, data = M_Mf)
anova(n_percent_aov)

#Carbon content 
c_percent_aov <- aov(log(c_percent) ~ fertilization*mowing*ditch*year, data = M_Mf)
 anova(c_percent_aov)

#pH
pH_aov <- aov(pH ~ fertilization*mowing*ditch*year, data = M_Mf)
anova(pH_aov)

#Bacterial community composition

#RICHNESS 
bccommunity_aov <- aov(bc_richness ~ fertilization*mowing*ditch*year, data = M_Mf)
anova(bccommunity_aov)

#SHANNON
bcshannon_aov <- aov(bc_shannon ~ fertilization*mowing*ditch*year, data = M_Mf)
anova(bcshannon_aov)

#INVSIMP
bcinvsimp_aov <- aov(bc_invsimp ~ fertilization*mowing*ditch*year, data = M_Mf)
anova(bcinvsimp_aov)

#Permaonva-- Mowed/Unmowed, Fertilized/Unfertilized
design_bac2 <- cbind(bc_diversity, otu_rel)

# Convert integers to characters
design_bac2$fertilization <- as.factor(design_bac2$fertilization)
design_bac2$mowing <- as.factor(design_bac2$mowing)
design_bac2$year <- as.factor(design_bac2$year)

adonis.all <- adonis(design_bac2[,-c(1:18)] ~ fertilization*ditch*year*mowing, method = "bray", data = design_bac2, perm=1000, set.seed=42)

adonis.all

#PERMANOVA---Mowed, Fertilized/Unfertilized
design_bac <- cbind(M_Mf, otu_rel)
design_bac1 <- design_bac[c(1:48),] #SUBSET FIRST 48 ROWS AND ALL COLUMNS B/C THEY ARE REPEATING

# Convert integers to characters
design_bac1$fertilization <- as.factor(design_bac1$fertilization)
design_bac1$mowing <- as.factor(design_bac1$mowing)
design_bac1$year <- as.factor(design_bac1$year)

adonis.141518 <- adonis(design_bac1[,-c(1:18)] ~ fertilization*ditch*year, method = "bray", data = design_bac1, perm=1000, set.seed=42)

adonis.141518



```
-------------------------------------------------------------------------------------------------------
#PLANTS (ONLY YEAR 2018)


*Diversity Metrics-- Richness, Shannon, Simpson's D*
```{r Diversity Metrics - Plant, echo= TRUE}
#CALCULATE PLANT RICHNESS
rownames(plant_M_MF_sort) = plant$sample
plant_sub <- plant_M_MF_sort[,-c(1:3)]

#TRANSFORM DATA TO 0/1 (ABSENCE/PRESENCE)
plant_PA <- decostand(plant_sub, method = "pa", na.rm=TRUE)

#COUNT THE NUMBER OF PRESENT(1) IN THE SAMPLE
plant_richness <- specnumber(plant_PA)
plant_richness

#ADD plant_richness TO df_plantdesign 
df_plantdesign <- cbind(df_plantdesign, plant_richness)

str(df_plantdesign)
df_plantdesign$ditch <- as.factor(df_plantdesign$ditch)
df_plantdesign$block <- as.factor(df_plantdesign$block)

#PLOT 
plant_rich <- ggplot(df_plantdesign, aes(x=treatment, y=plant_richness, color=as.factor(treatment)))+ geom_boxplot()+
geom_point(aes(color=factor(treatment), shape=ditch), size=2, position = position_jitterdodge())+ scale_color_manual(name="Treatment", values=c("peru", "darkmagenta","gray70", "darkgreen"), labels = c("Unmowed/Unfert","Unmowed/Fert", "Mowed/Unfert", "Mowed/Fertilized"))+
xlab("Treatment")+
ylab("Plant Richness (OTUs)")+
scale_shape_manual(labels = c("(0_wet)","(1_dry)"), values= c(16,17))+
theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm"))+ 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1))
plant_rich

#Save plot 
ggsave("C:/Users/ohhey/OneDrive/Pictures/plant_rich.png", plot=plant_rich, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)


#PLOT BY YEAR----2015 AND 2018 IS INCLUDED...THOUGHT WE ONLY HAD DATA FOR 2018...WILL CHECK ON THIS
ggplot(data=soil_div) +
geom_boxplot(aes(x=treatment, y=plant_richness, fill=treatment), na.rm = TRUE)+
geom_point(aes(x=treatment, y=plant_richness, color=as.factor(ditch)), position=position_jitterdodge())+
scale_color_manual(values = c("red", "blue", "gray"))+
scale_fill_manual(values=c("grey", "green", "yellow", "cyan"))+
  facet_wrap(~year)



#CALCULATE SHANNON H'
plant_shannon <- diversity(plant_sub, "shannon")
plant_shannon
 
#ADD SHANNON TO df_plantdesign DATAFRAME
df_plantdesign <- cbind(df_plantdesign, plant_shannon)


str(df_plantdesign)
df_plantdesign$ditch <- as.factor(df_plantdesign$ditch)
df_plantdesign$block <- as.factor(df_plantdesign$block)

#PLOT
plant_shannon <- ggplot(df_plantdesign, aes(x=treatment, y=plant_richness, color=as.factor(treatment)))+ geom_boxplot()+
geom_point(aes(color=factor(treatment), shape=ditch), size=2, position = position_jitterdodge())+ scale_color_manual(name="Treatment", values=c("peru", "darkmagenta","gray70", "darkgreen"), labels = c("Unmowed/Unfert","Unmowed/Fert", "Mowed/Unfert", "Mowed/Fertilized"))+
xlab("Treatment")+
ylab("Plant Shannon Diversity (H')")+
scale_shape_manual(labels = c("(0_wet)","(1_dry)"), values= c(16,17))+
theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm"))+ 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1))
plant_shannon

#Save plot 
ggsave("C:/Users/ohhey/OneDrive/Pictures/plant_shannon.png", plot=plant_shannon, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)

#SHANNON BY YEAR
ggplot(data=soil_div) +
geom_boxplot(aes(x=treatment, y=plant_shannon, fill=treatment), na.rm = TRUE)+
geom_point(aes(x=treatment, y=plant_shannon, color=as.factor(ditch)), position=position_jitterdodge())+
scale_color_manual(values = c("red", "blue", "gray"))+
scale_fill_manual(values=c("grey", "green", "yellow", "cyan"))+
  facet_wrap(~year)


#CALCULATING INVERSE SIMPSONS D
plant_invsimp <-diversity(plant_sub, "invsimpson")
plant_invsimp

#ADD INVERSE SIMPSONS D TO df_plantdesign DATAFRAME
df_plantdesign <- cbind(df_plantdesign, plant_invsimp)

str(df_plantdesign)
df_plantdesign$ditch <- as.factor(df_plantdesign$ditch)
df_plantdesign$block <- as.factor(df_plantdesign$ditch)

#PLOT
plant_invsimp <- ggplot(df_plantdesign, aes(x=treatment, y=plant_invsimp, color=as.factor(treatment)))+ geom_boxplot()+
geom_point(aes(color=factor(treatment), shape=ditch), size=2, position = position_jitterdodge())+ scale_color_manual(name="Treatment", values=c("peru", "darkmagenta","gray70", "darkgreen"), labels = c("Unmowed/Unfert","Unmowed/Fert", "Mowed/Unfert", "Mowed/Fertilized"))+
xlab("Treatment")+
ylab("Plant Simpson's Index (D)")+
scale_shape_manual(labels = c("(0_wet)","(1_dry)"), values= c(16,17))+
theme_bw() + 
    theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line 
          =element_line(colour = "black")) + 
    theme(axis.title=element_text(vjust=1,size=14,face="bold"),
          axis.text=element_text(size=14), axis.text.x = element_text(vjust=0.65, hjust=0.5, 
          size=14), panel.border = element_rect(colour = "black",size=1)) + 
    theme(axis.ticks.length=unit(0.3,"cm"))+ 
    theme(strip.text.x = element_text(size=14, face="bold"), strip.text.y =   
          element_text(size=14, face="bold"), strip.background = element_rect(colour="black",
          fill="white", size=1))
plant_invsimp

#Save plot 
ggsave("C:/Users/ohhey/OneDrive/Pictures/plant_invsimp.png", plot=plant_invsimp, device=NULL, path=NULL, scale=1, width=6.3, height=4.4, dpi=900, limitsize=TRUE)

#SIMPSONS D BY YEAR
ggplot(data=soil_div) +
geom_boxplot(aes(x=treatment, y=plant_invsimp, fill=treatment), na.rm = TRUE)+
geom_point(aes(x=treatment, y=plant_invsimp, color=as.factor(ditch)), position=position_jitterdodge())+
scale_color_manual(values = c("red", "blue", "gray"))+
scale_fill_manual(values=c("grey", "green", "yellow", "cyan"))+
  facet_wrap(~year)

```
*CALCULATE DISTANCE MATRIX AND PCoA COMPONENTS*
```{r Ordination (PCoA)- Plant, include= FALSE}
#BUILD THE MATRIX
plant_dist <-vegdist(plant_sub, method="bray")

#CALCULATE PCoA VALUES
plant_dis_pcoa <- cmdscale(plant_dist, k=3, eig=TRUE, add=FALSE)

#CLASSICAL (METRIC) MULTIDIMENSIONAL SCALING
explainvar1a <- round(plant_dis_pcoa$eig[1] / sum(plant_dis_pcoa$eig), 3) * 100
explainvar2a <- round(plant_dis_pcoa$eig[2] / sum(plant_dis_pcoa$eig), 3) * 100
sum.eiga <- sum(explainvar1a, explainvar2a)
explainvar1a #36.4%
explainvar2a #21.6%

#PCoA BASED ON PLANTS (DO NOT EDIT)
pcoa.groups <- paste(df_plantdesign$ditch, df_plantdesign$treatment, sep = "_")
pcoa.points <- data.frame(plant_dis_pcoa$points, group = pcoa.groups)

#Set Std Err and Conf Int
se <- function(x, ...) {
  sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))
}

#Calculate Centroids (mean and SE)
pcoa.L.centroids <- melt(pcoa.points, id="group", measure.vars = c("X1", "X2"))
pcoa.centroids <- acast(pcoa.L.centroids, variable ~ group, mean)
pcoa.centroids.se <- acast(pcoa.L.centroids, variable ~ group, se)
pcoa.centroids.sd <- acast(pcoa.L.centroids, variable ~ group, sd)

#Combine
pcoa.cent.dataframe <- cbind(t(pcoa.centroids), t(pcoa.centroids.se))
colnames(pcoa.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")
pcoa.cent.treats <- rownames(pcoa.cent.dataframe)

pcoa.col <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 2)) # treatment
pcoa.shape <- as.factor(sapply(strsplit(pcoa.cent.treats, "_"), `[`, 1))  # source

levels(pcoa.col) <- c("Unmowed/Unfert","Unmowed/Fert","Mowed/Unfert","Mowed-Fert")

#PLOT PCoA
df1a <- as.data.frame(pcoa.cent.dataframe)
plot1 <- ggplot(df1a, aes(x=V1, y=V2), group = interaction(pcoa.col, pcoa.shape)) + 
  theme_bw() +
  geom_point(aes(shape = pcoa.shape, colour = pcoa.col), size=10, stroke = 2) + 
  #geom_point(aes(colour=pcoa.col2), size =4.0)+
  #removes gridlines from plot
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(), 
               axis.line = element_line(colour = "black"))+ 
  #Set error bars for geom_point 
geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0), colour="black") + geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0), colour="black") +
  #Set colors for treatments
  scale_colour_manual(values = c("peru","darkmagenta","gray70","darkgreen")) +
  scale_shape_manual(values = c(16,22,15)) +
  #Sets map coordinates
  coord_cartesian(xlim = c(-0.4, 0.4), ylim = c(-0.4, 0.4)) +
  #Sets axis text and put border around plot
  theme(axis.title = element_text(size=14),# face="bold"), 
        axis.text.x = element_text(size=14, color="black"),  axis.text.y = element_text(size=14, color="black"),
        panel.border = element_rect(colour = "black", size=1.25)) +
  #Set plot title text size
  theme(plot.title=element_text(size=14)) +
  #Set legend text size
  theme(legend.text=element_text(size=14, face="bold"), legend.title = element_text(size=14, face="bold"))+
  #Sets size of tick marks on axis
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  #Sets labels for plot title, axis titles, and legend headings
  xlab("PCoA 1 (36.4%)") + ylab("PCoA 2 (21.6%)") + 
  labs(shape = "Ditch") +
  labs(colour="Treatment") +
  ggtitle("Plant community beta diveristy")+
  theme(legend.position="right")
  
plot1 
```






